src/Model/Scope/BasicCondition.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model\Scope;
+
+use atk4\core\ReadableCaptionTrait;
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\dsql\Expression;
+use atk4\dsql\Expressionable;
+
+class BasicCondition extends AbstractCondition
+{
+    use ReadableCaptionTrait;
+
+    public $key;
+
+    public $operator;
+
+    public $value;
+
+    protected static $opposites = [
+        '=' => '!=',
+        '!=' => '=',
+        '<' => '>=',
+        '>' => '<=',
+        '>=' => '<',
+        '<=' => '>',
+        'LIKE' => 'NOT LIKE',
+        'NOT LIKE' => 'LIKE',
+        'IN' => 'NOT IN',
+        'NOT IN' => 'IN',
+        'REGEXP' => 'NOT REGEXP',
+        'NOT REGEXP' => 'REGEXP',
+    ];
+
+    protected static $dictionary = [
+        '=' => 'is equal to',
+        '!=' => 'is not equal to',
+        '<' => 'is smaller than',
+        '>' => 'is greater than',
+        '>=' => 'is greater or equal to',
+        '<=' => 'is smaller or equal to',
+        'LIKE' => 'is like',
+        'NOT LIKE' => 'is not like',
+        'IN' => 'is one of',
+        'NOT IN' => 'is not one of',
+        'REGEXP' => 'is regular expression',
+        'NOT REGEXP' => 'is not regular expression',
+    ];
+
+    protected static $skipValueTypecast = [
+        'LIKE',
+        'NOT LIKE',
+        'REGEXP',
+        'NOT REGEXP',
+    ];
+
+    public function __construct($key, $operator = null, $value = null)
+    {
+        if (func_num_args() == 2) {
+            $value = $operator;
+            $operator = '=';
+        }
+
+        if (is_bool($key)) {
+            if ($key) {
+                return;
+            }
+
+            $key = new Expression('false');
+        }
+
+        $this->key = $key;
+        $this->operator = $operator;
+        $this->value = $value;
+    }
+
+    public function onChangeModel(): void
```

Comment:
why not setModel() ?

Reply:
`setModel` implies we are setting a model but actually we are not. Just performing actions when model is changed.

Reply:
> 
> 
> `setModel` implies we are setting a model but actually we are not. Just performing actions when model is changed.
> Model is only set in the `Model\Scope` object (in `$model->scope` property)

so `if ($this->model !== $model) {` condition in `setModel()`?

Reply:
done